src/Renderer/LayeredMaterial.js

```diff
@@ -139,218 +82,210 @@ const LayeredMaterial = function LayeredMaterial(options) {
+    }
 
-    for (let l = 0; l < layerTypesCount; l++) {
-        for (let i = 0, max = this.textures[l].length; i < max; i++) {
-            if (this.textures[l][i] instanceof THREE.Texture) {
-                this.textures[l][i].dispose();
+    // release texture atlas
+    for (const atlas of this.uniforms.atlasTextures.value) {
+        if (atlas.id != emptyAtlas.id) {
+            atlas._ownerCount--;
+            if (atlas._ownerCount == 0) {
+                atlas.dispose();
             }
         }
     }
 };
 
-LayeredMaterial.prototype.setSequence = function setSequence(sequenceLayer) {
-    let offsetLayer = 0;
-    let offsetTexture = 0;
+function _swapIndices(array, newIndices) {
+    const copy = [...array];
+    for (const pair of newIndices) {
+        array[pair.new] = copy[pair.old];
+    }
+}
 
-    const originalOffsets = new Array(...this.uniforms.offsetScale_L01.value);
-    const originalTextures = new Array(...this.uniforms.dTextures_01.value);
+LayeredMaterial.prototype.setSequence = function setSequence(sequenceLayer, updateFragmentShader) {
+    const newIndices = [];
 
+    let offset = 0;
+    // assign new index
     for (let l = 0; l < sequenceLayer.length; l++) {
         const layer = sequenceLayer[l];
         const oldIndex = this.indexOfColorLayer(layer);
         if (oldIndex > -1) {
-            const newIndex = l - offsetLayer;
-            const texturesCount = this.layerTexturesCount[oldIndex];
-
-            // individual values are swapped in place
-            if (newIndex !== oldIndex) {
-                moveElementArray(this.colorLayersId, oldIndex, newIndex);
-                moveElementArray(this.layerTexturesCount, oldIndex, newIndex);
-                moveElementArray(this.uniforms.paramLayers.value, oldIndex, newIndex);
-                moveElementArray(this.uniforms.visibility.value, oldIndex, newIndex);
-            }
-            const oldOffset = this.getTextureOffsetByLayerIndex(newIndex);
-            // consecutive values are copied from original
-            for (let i = 0; i < texturesCount; i++) {
-                this.uniforms.offsetScale_L01.value[offsetTexture + i] = originalOffsets[oldOffset + i];
-                this.uniforms.dTextures_01.value[offsetTexture + i] = originalTextures[oldOffset + i];
-            }
-
-
-            this.setTextureOffsetByLayerIndex(newIndex, offsetTexture);
-            offsetTexture += texturesCount;
+            const newIndex = l - offset;
+            newIndices.push({ new: newIndex, old: oldIndex });
         } else {
-            offsetLayer++;
+            offset++;
         }
     }
 
+    // build new arrays
+    _swapIndices(this.colorLayersId, newIndices);
+    _swapIndices(this.uniforms.paramLayers.value, newIndices);
+    _swapIndices(this.uniforms.atlasTextures.value, newIndices);
+
     this.uniforms.colorLayersCount.value = this.getColorLayersCount();
+
+    if (updateFragmentShader || updateFragmentShader == undefined) {
```

Comment:
Looks like you want to use a [default parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters) value for `updateFragmentShader` instead.